src/dentry.c

```diff
@@ -846,7 +851,7 @@ get_parent_dentry(WIMStruct *wim, const tchar *path,
 		  CASE_SENSITIVITY_TYPE case_type)
 {
 	size_t path_len = tstrlen(path);
-	tchar buf[path_len + 1];
+	smart_array(tchar, buf, path_len + 1);
```

Comment:
I'd just drop the `smart_array()` call altogether and add an explicit `alloca()`, so that anybody who reads the code realises that this actually issues a memory allocation call, regardless of the compiler, as I think these eventually should be replaced by formal MALLOC()/FREE() calls to actually use the wimlib memory allocation overrides -- but of course that is unrelated to the MSVC porting effort.